iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/executor/RegionWriteExecutor.java

```diff
@@ -242,24 +242,27 @@ public RegionExecutionResult visitPipeEnrichedInsertNode(
     private RegionExecutionResult executeDataInsert(
         InsertNode insertNode, WritePlanNodeExecutionContext context) {
       RegionExecutionResult response = new RegionExecutionResult();
-      context.getRegionWriteValidationRWLock().readLock().lock();
+
       try {
+        context.getRegionWriteValidationRWLock().readLock().lock();
+
         TSStatus status = fireTriggerAndInsert(context.getRegionId(), insertNode);
         response.setAccepted(TSStatusCode.SUCCESS_STATUS.getStatusCode() == status.getCode());
         response.setMessage(status.message);
         if (!response.isAccepted()) {
           response.setStatus(status);
         }
-        return response;
-      } catch (ConsensusException e) {
+      } catch (NullPointerException | ConsensusException e) {
```

Comment:
I think it's better to check whether `Lock` is null independently, because `fireTriggerAndInsert` is a complicate function and catch NPE may hidden some other mistake.

Reply:
I think it is ok. It will log a warning and retry for other possible NPE.

Reply:
do not catch npe,it's ugly.

@liyuheng55555 you can just fix this as we discussed before~

Reply:
ok, I will close this pr